src/libraries/Microsoft.Extensions.DependencyInjection/src/Resources/Strings.resx

```diff
@@ -180,4 +180,10 @@
   <data name="TrimmingAnnotationsDoNotMatch_NewConstraint" xml:space="preserve">
     <value>Generic implementation type '{0}' has a DefaultConstructorConstraint ('new()' constraint), but the generic service type '{1}' doesn't.</value>
   </data>
-</root>
+  <data name="AotCannotCreateEnumerableValueType" xml:space="preserve">
+    <value>Unable to create an Enumerable service of type '{0}' because it is a ValueType. ValueType enumerable services will not work in AOT applications.</value>
+  </data>
+  <data name="AotCannotCreateGenericValueType" xml:space="preserve">
+    <value>Unable to create a generic service for type '{0}' because '{1}' is a ValueType. Generic services with ValueTypes will not work in AOT applications.</value>
+  </data>
```

Comment:
Not sure if this is an improvement, but throwing this in:

I think it might be useful to keep referring to the no-compromise AOT mode as "native AOT" to distinguish it from the more forgiving AOT modes we have.

```suggestion
  <data name="AotCannotCreateEnumerableValueType" xml:space="preserve">
    <value>Unable to create an Enumerable service of type '{0}' because it is a ValueType. Native code to support creating Enumerable services might not be available with native AOT.</value>
  </data>
  <data name="AotCannotCreateGenericValueType" xml:space="preserve">
    <value>Unable to create a generic service for type '{0}' because '{1}' is a ValueType. Native code to support creating generic services might not be available with native AOT</value>
  </data>
```